core/embed/rust/src/ui/layout_eckhart/flow/continue_recovery_homepage.rs

```diff
@@ -206,94 +184,187 @@ pub fn new_continue_recovery_homepage(
+) {
+    match recovery_type {
+        RecoveryType::Normal if show_instructions => (
+            Header::new(TR::recovery__title.into()).with_menu_button(),
+            TR::buttons__continue.into(),
+            TR::recovery__title_cancel_recovery.into(),
+            TR::recovery__title.into(),
+            TR::recovery__wanna_cancel_recovery.into(),
+        ),
+        RecoveryType::Normal => (
+            Header::new(TR::words__title_done.into())
+                .with_text_style(theme::label_title_confirm())
+                .with_icon(theme::ICON_DONE, theme::GREEN_LIGHT)
+                .with_menu_button(),
+            TR::instructions__enter_next_share.into(),
+            TR::recovery__title_cancel_recovery.into(),
+            TR::recovery__title.into(),
+            TR::recovery__wanna_cancel_recovery.into(),
+        ),
+        // 1st screen of the recovery process
+        _ if show_instructions => (
+            Header::new(TR::reset__check_wallet_backup_title.into()).with_menu_button(),
+            TR::buttons__continue.into(),
+            TR::buttons__cancel.into(),
+            TR::reset__check_wallet_backup_title.into(),
+            TR::recovery__wanna_cancel_dry_run.into(),
+        ),
+        _ => (
+            // multi-group recovery check
+            if subtext.is_none() {
+                Header::new(TR::reset__check_wallet_backup_title.into()).with_menu_button()
+            // single-group recovery check
+            } else {
+                Header::new(TR::words__title_done.into())
+                    .with_text_style(theme::label_title_confirm())
+                    .with_icon(theme::ICON_DONE, theme::GREEN_LIGHT)
+                    .with_menu_button()
+            },
+            TR::buttons__continue.into(),
+            TR::buttons__cancel.into(),
+            TR::reset__check_wallet_backup_title.into(),
+            TR::recovery__wanna_cancel_dry_run.into(),
+        ),
+    }
+}
+
+fn flow_before_shares(
+    content_main: MsgMap<
+        SendButtonRequest<TextScreen<Paragraphs<ParagraphVecShort<'static>>>>,
+        impl Fn(TextScreenMsg) -> Option<FlowMsg> + 'static,
+    >,
+    cancel_btn: TString<'static>,
+) -> Result<SwipeFlow, error::Error> {
+    let content_menu = VerticalMenuScreen::new(VerticalMenu::<ShortMenuVec>::empty().with_item(
+        Button::new_menu_item(cancel_btn, theme::menu_item_title_orange()),
+    ))
+    .with_header(Header::new(TString::empty()).with_close_button())
+    .map(|msg| match msg {
+        VerticalMenuScreenMsg::Selected(i) => Some(FlowMsg::Choice(i)),
+        VerticalMenuScreenMsg::Close => Some(FlowMsg::Cancelled),
+        _ => None,
+    });
+
+    let mut res = SwipeFlow::new(&ContinueRecoveryBeforeShares::Main)?;
+    res.add_page(&ContinueRecoveryBeforeShares::Main, content_main)?
+        .add_page(&ContinueRecoveryBeforeShares::Menu, content_menu)?;
+    Ok(res)
+}
+
+fn flow_between_shares_simple(
+    content_main: MsgMap<
```

Comment:
Those types a quite verbose... I'll open a separate issue to refactor the API so in order to avoid them :)